loki/transformations/transform_loop.py

```diff
@@ -752,3 +753,59 @@ def visit_Loop(self, o, *args, **kwargs):
 
     with pragmas_attached(routine, Loop):
         routine.body = PragmaLoopUnrollTransformer(warn_iterations_length=warn_iterations_length).visit(routine.body)
+
+
+class TransformLoopsTransformation(Transformation):
+    """
+    A :any:`Transformation` that provides a common location for the various loop transformations to be called
+    in a :any:`Scheduler` pipeline.
+
+    The transformation applies the following methods in order:
+
+    * :any:`loop_interchange`
+    * :any:`loop_fusion`
+    * :any:`loop_fission`
+    * :any:`loop_unroll`
+
+    Parameters
+    ----------
+    project_bounds : bool
+        Project loop bounds whilst performing loop interchange. Default: ``False``.
+    promote : bool
+        Try to automatically detect read-after-write across fission points
+        and promote corresponding variables. Note that this does not affect
+        promotion of variables listed directly in the pragma's ``promote``
+        option. Default: ``True``.
+    warn_loop_carries : bool
+        Try to automatically detect loop-carried dependencies and warn
+        when the fission point sits after the initial read and before the
+        final write. Default: ``True``.
+    warn_iterations_length : bool
+        This specifies if warnings should be generated when unrolling
+        loops with a large number of iterations (32). It's mainly to
+        disable warnings when loops are being unrolled for internal
+        transformations and analysis. Default: ``True``.
+    """
+
+    def __init__(
+            self, project_bounds=False, promote=True, warn_loop_carries=True,
+            warn_iterations_length=True
```

Comment:
Without the loop transformation context these options are a little hard to map mentally. I suggest we prefix them according to the transformation they belong to:
```suggestion
            self, interchange_project_bounds=False,
            fission_auto_promote=True, fission_warn_loop_carries=True,
            unroll_warn_iterations_length=True
```